From c7f81bab72a788f5bfaa39775f42483d19bce170 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 5 Dec 2016 18:03:32 +0800
Subject: [PATCH 1373/5242] MLK-13607-5 ARM: dts: imx7ulp-evk: add mipi dsi
 support

commit  93f6a8705fed9f7c6ff8500ab8f31e6596c8ad50 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add dts support for mipi dsi module on imx7ulp-evk board.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 51584bcf07c6e97631bd22ed7eeb8bd14cbfc61a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   22 ++++++++++++++++++++++
 arch/arm/boot/dts/imx7ulp.dtsi    |   10 ++++++++++
 2 files changed, 32 insertions(+)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 8b27d61..ffb3240 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -40,6 +40,13 @@
 		status = "okay";
 	};
 
+	mipi_dsi_reset: mipi-dsi-reset {
+		compatible = "gpio-reset";
+		reset-gpios = <&gpio0 19 GPIO_ACTIVE_LOW>;
+		reset-delay-us = <1000>;
+		#reset-cells = <0>;
+	};
+
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
@@ -181,6 +188,12 @@
 			>;
 		};
 
+		pinctrl_mipi_dsi_reset: mipi_dsi_reset_grp {
+			fsl,pins = <
+				IMX7ULP_PAD_PTC19__PTC19		0x20103
+			>;
+		};
+
 		pinctrl_lpuart4: lpuart4grp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTC3__LPUART4_RX	0x400
@@ -288,6 +301,7 @@
 
 &lcdif {
 	status = "okay";
+	disp-dev = "mipi_dsi_northwest";
 	display = <&display0>;
 
 	display0: display {
@@ -360,6 +374,14 @@
 	};
 };
 
+&mipi_dsi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_mipi_dsi_reset>;
+	lcd_panel = "TRULY-WVGA-TFT3P5581E";
+	resets = <&mipi_dsi_reset>;
+	status = "okay";
+};
+
 &lpuart4 { /* console */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_lpuart4>;
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index 30c6971..a745a92 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -452,6 +452,16 @@
 			status = "disabled";
 		};
 
+		mipi_dsi: mipi_dsi@40A90000 {
+			compatible = "fsl,imx7ulp-mipi-dsi";
+			reg = <0x40A90000 0x10000>;
+			interrupts = <GIC_SPI 46 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clks IMX7ULP_CLK_DSI>;
+			clock-names = "mipi_dsi_clk";
+			sim = <&sim>;
+			status = "disabled";
+		};
+
 		mmdc: mmdc@40ab0000 {
 			compatible = "fsl,imx7ulp-mmdc";
 			reg = <0x40ab0000 0x4000>;
-- 
1.7.9.5

